sdc/str_arr_ext.py

```diff
@@ -688,12 +692,22 @@ def str_arr_from_list(in_list):
         total_chars = 0
         # TODO: use vector to avoid two passes?
         # get total number of chars
-        for s in in_list:
-            total_chars += get_utf8_size(s)
+        nan_mask = np.zeros(n_strs, dtype=np.bool_)
+        for i in numba.prange(n_strs):
+            s = in_list[i]
+            if s is None:
+                nan_mask[i] = True
+            else:
+                total_chars += get_utf8_size(s)
 
         A = pre_alloc_string_array(n_strs, total_chars)
-        for i in range(n_strs):
-            A[i] = in_list[i]
+
+        for i in numba.prange(n_strs):
+            if nan_mask[i]:
+                A[i] = ''
+            else:
+                A[i] = in_list[i]
```

Comment:
Seems It looks better in a single line: `A[i] = '' if nan_mask[i] else in_list[i]`

Reply:
Right, thanks!